.github/workflows/java-cd.yml

```diff
@@ -1,77 +1,78 @@
-          required: true
+        inputs:
+            version:
+                description: 'The release version of GLIDE, formatted as *.*.* or *.*.*-rc*'
+                required: true
+            maven_publish:
+                description: 'Publish to Maven Central'
+                required: true
+                type: boolean
 
 concurrency:
     group: java-cd-${{ github.head_ref || github.ref }}
     cancel-in-progress: true
 
 permissions:
-      id-token: write
+    id-token: write
 
 jobs:
-    start-self-hosted-runner:
-        if: github.repository_owner == 'valkey-io'
+    load-platform-matrix:
         runs-on: ubuntu-latest
         environment: AWS_ACTIONS
+        outputs:
+            PLATFORM_MATRIX: ${{ steps.load-platform-matrix.outputs.PLATFORM_MATRIX }}
         steps:
-          - name: Checkout
-            uses: actions/checkout@v4
-            with:
-                fetch-depth: 0
-
-          - name: Start self hosted EC2 runner
-            uses: ./.github/workflows/start-self-hosted-runner
-            with:
-                role-to-assume: ${{ secrets.ROLE_TO_ASSUME }}
-                aws-region: ${{ secrets.AWS_REGION }}
-                ec2-instance-id: ${{ secrets.AWS_EC2_INSTANCE_ID }}
+            - name: Checkout
+              uses: actions/checkout@v4
+
+            - name: load-platform-matrix
+              id: load-platform-matrix
+              shell: bash
+              run: |
+                  # Get the matrix from the matrix.json file, without the object that has the IMAGE key
+                  export "PLATFORM_MATRIX=$(jq 'map(select(.PACKAGE_MANAGERS | contains(["maven"])))' < .github/json_matrices/build-matrix.json | jq -c .)"
+                  echo "PLATFORM_MATRIX=${PLATFORM_MATRIX}" >> $GITHUB_OUTPUT
+
+    set-release-version:
+        needs: load-platform-matrix
+        runs-on: ubuntu-latest
+        outputs:
+            RELEASE_VERSION: ${{ steps.release-version.outputs.RELEASE_VERSION }}
+        steps:
+            - name: Set the release version
+              id: release-version
+              shell: bash
+              run: |
+                if ${{ github.event_name == 'workflow_dispatch' }}; then
+                    R_VERSION="${{ env.INPUT_VERSION }}"
+                else
+                    R_VERSION=${GITHUB_REF:11}
+                fi
+                    echo "RELEASE_VERSION=${R_VERSION}" >> $GITHUB_ENV
+                    echo "Release version detected: $R_VERSION"
+                echo "RELEASE_VERSION=$R_VERSION" >> $GITHUB_OUTPUT
+              env:
+                EVENT_NAME: ${{ github.event_name }}
+                INPUT_VERSION: ${{ github.event.inputs.version }}
 
     create-binaries-to-publish:
-        needs: start-self-hosted-runner
+        needs: [set-release-version, load-platform-matrix]
         if: github.repository_owner == 'valkey-io'
         timeout-minutes: 35
         env:
             JAVA_VERSION: '11'
```

Comment:
You probably don't this env var, since it is hardcoded (I don't think that we'll matrix it)

Reply:
Both are needed to be there or we can't access `needs.<job>.outputs.<variable>`.